lib/python/test/testsuite/Math/physics.py

```diff
@@ -0,0 +1,219 @@
+"""
+
+__all__ = ["calculateV_O",
+           "calculateTime",
+           "calculateTimeFreq",
+           "plasmafrequence",
+           "calculateBeta"]
+
+import numpy as np
+from scipy.constants import c, epsilon_0, e, m_e
+from . import _searchData as sD
+
+
+def calculateV_O(gamma=None):
+    """
+    calculate the speed v and use
+
+    gamma  = 1 / (1 - beta**2)**0.5
+
+    Input:
+    -------
+    gamma :   float or numpy.ndarray, optional
+              lorentz factor
+              If None, gamma is searched for in the parameter files.
+
+    Return:
+    -------
+    out :    float or numpy.ndarray
+             speed given by gamma
+    """
+
+    if gamma is None:
+        gamma = sD.searchParameter("gamma",
+                                   directiontype="param")
+
+    return np.sqrt((1 - 1 / gamma**2) * c ** 2)
+
+
+def calculateTimeFreq(frequency,
+                      steps=None,
+                      deltaT: float = None,
+                      **kwargs):
+    """
+    calculate the time in [frequency ** -1]
+
+    Input:
+    -------
+    steps :     list, optional
+                list of number of timesteps
+                if None, steps are searched for in the .dat files
+
+    deltaT :    float, optional
+                time step of the simulation
+                if None, deltaT is searched for in the parameter files
+
+    frequency : float
+                plasma frequency
+
+    Return:
+    -------
+    out : list
+          time
+    """
+
+    if steps is None:
+        steps = sD.searchParameter("step",
+                                   directiontype="dat",
+                                   **kwargs)
+
+    if deltaT is None:
+        deltaT = sD.searchParameter("DELTA_T_SI",
+                                    directiontype="param")
+    return steps * deltaT * frequency
+
+
+def calculateTime(steps=None, deltaT: float = None):
+    """
+    calculate the time in s
+
+    Input:
+    -------
+    steps :     list
+                list of number of timesteps
+                if None, steps are searched for in the .dat files
+
+    deltaT :    float
+                time step of the simulation
+                if None, deltaT is searched for in the parameter files
+
+    Return:
+    -------
+    out : list
+          time
+    """
+    if steps is None:
+        steps = sD.searchParameter("step",
+                                   directiontype="dat")
+
+    if deltaT is None:
+        deltaT = sD.searchParameter("DELTA_T_SI",
+                                    directiontype="param")
+
+    return steps * deltaT
+
+
+def calculateBeta(v_0=None):
+    """
+    calculate beta with
+
+        beta = v_0 / c
+
+    Input:
+    -------
+    v_0:    float or numpy.ndarray
+            speed
+
+    Use:
+    -------
+    calculateV_O() if v_0 is None
+
+    Return:
+    -------
+    beta:    float or numpy.ndarray
+    """
+
+    if v_0 is None:
+        v_0 = calculateV_O()
+
+    return v_0 / c
```

Comment:
wouldn't it bet better to implement beta and than derive v_0 from beta? 
In v_0 you need to multiply with c. And here you have to divide by c again.